components/dm/command/import.go

```diff
@@ -0,0 +1,109 @@
+package command
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/fatih/color"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tiup/components/dm/ansible"
+	"github.com/pingcap/tiup/pkg/cliutil"
+	"github.com/pingcap/tiup/pkg/cluster"
+	cansible "github.com/pingcap/tiup/pkg/cluster/ansible"
+	tiuputils "github.com/pingcap/tiup/pkg/utils"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+)
+
+func newImportCmd() *cobra.Command {
+	var ansibleDir string
+	var inventoryFileName string
+	var rename string
+	var clusterVersion string
+
+	cmd := &cobra.Command{
+		Use:   "import",
+		Short: "Import an exist DM 1.0 cluster from tidb-ansible and re-deploy 2.0 version",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := supportVersion(clusterVersion); err != nil {
+				return errors.AddStack(err)
+			}
+
+			importer, err := ansible.NewImporter(ansibleDir, inventoryFileName)
+			if err != nil {
+				return errors.AddStack(err)
+			}
+
+			clusterName, meta, err := importer.ImportFromAnsibleDir()
+			if err != nil {
+				return errors.AddStack(err)
+			}
+
+			if rename != "" {
+				clusterName = rename
+			}
+
+			err = importer.ScpSourceToMaster(meta.Topology)
+			if err != nil {
+				return errors.AddStack(err)
+			}
+
+			data, err := yaml.Marshal(meta.Topology)
+			if err != nil {
+				return errors.AddStack(err)
+			}
+
+			f, err := ioutil.TempFile("", "tiup-*")
```

Comment:
How about giving it a meaningful name? like `tiup-source-{source-id}.yaml`?

Reply:
This is the temporal topology.yaml not about source  config of dm.

Reply:
I want to comment on this https://github.com/pingcap/tiup/pull/672#pullrequestreview-466687108. 😢